src/main/java/pl/jablonskanycz/bakery/clients/FileBasedClientRepository.java

```diff
@@ -1,30 +1,127 @@
+                    return new Client(
+                        Integer.parseInt(client[0]),
+                        client[1],
+                        client[2],
+                        addressRepository.findById(Integer.parseInt(client[3])));
+                })
+                .collect(Collectors.toList());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return clients;
     }
 
     @Override
     public Client findBySurname(String surname) {
-        return null;
+        Client client = null;
+        try {
+            return Files.lines(clientPath)
+                    .skip(1)
+                    .map(line -> {
+                        String[] strings = line.split(",");
+                        return new Client(
+                                Integer.parseInt(strings[0]),
+                                strings[1],
+                                strings[2],
+                                addressRepository.findById(Integer.parseInt(strings[3])));
+                    })
+                    .filter(a -> surname.equals(a.getSurname()))
+                    .findFirst()
+                    .orElseThrow(() -> new NoSuchElementException("Client list is empty"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return client;
     }
 
     @Override
     public void addClient(Client clientToAdd) {
-
+        List<String> clientsList = null;
+        try {
+            clientsList = Files.lines(clientPath)
+                    .toList();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        String lastId = clientsList.get(clientsList.size() - 1).split(",")[0];
+        int newId = Integer.parseInt(lastId) + 1;
+        String line = "\n" + newId + "," + clientToAdd.getName() + "," + clientToAdd.getSurname() + "," + clientToAdd.getAddress();
+        try {
+            Files.writeString(clientPath, line, APPEND);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
 
     @Override
-    public void updateClient(Client clientToUpdate) {
-
+    public void updateClient(Client clientWithOldData, Client clientWithNewData) { //jak zrobić overwrite jednej linii?
+        try {
+            Optional<Client> client = Files.lines(clientPath)
+                    .skip(1)
+                    .map(line -> {
+                        String[] strings = line.split(",");
+                        return new Client(
+                                Integer.parseInt(strings[0]),
+                                strings[1],
+                                strings[2],
+                                addressRepository.findById(Integer.parseInt(strings[3])));
+                    })
+                    .filter(c -> c.equals(clientWithOldData))
+                    .map(c -> {
+                        c.setId(clientWithNewData.getId());
```

Comment:
I would suggest to extract the creation of a new `client` into a method, for a better readibility :)